src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -153,34 +158,46 @@ public ResourceEnum getCacheType() {
     private Long currentCacheMaxSizeInBytes;
     private Long desiredCacheMaxSizeInBytes;
     private Long heapMaxSizeInBytes;
+    private final long upperBoundInBytes;
+    private final long lowerBoundInBytes;
 
     private Builder(
         final NodeKey esNode,
         final ResourceEnum cacheType,
         final AppContext appContext,
-        double upperBoundThreshold) {
+        final RcaConf conf) {
```

Comment:
maybe it would be better to pass CacheActionConfig in builder ?

Reply:
That was the original thought. But I went with this approach for the following reasons -

1. Consumers should not have to handle which config to pass to which object. The object should be able to self extract its own relevant config. 
2. Wanted to keep doors open for the action to consume configs outside of `CacheActionConfig` if needed in future.

Passing a subset config only makes it harder to do things with the config going forward (including refactoring). e.g. deciders holding only `DeciderConfig` made it impossible to pass action specific config to actions. Just passing the parent everywhere and letting consumers extract what they need made sense to me.